### DISCUSSION TITLE
[@types/lodash] cloneDeep should remove the readonly attribute

### DISCUSSION THREAD

**Discussion by air2** [DefinitelyTyped/DefinitelyTyped#70545] [Issues with a @types package]
So I have:
```
const source: ReadonlyArray<string> = ['data']
const destination =  cloneDeep(source)
destination.push('item')
```
This is not allowed because cloneDeep clones the type exactly, IMHO it should not, it should remove the readonly from the cloned type.

[a comment by an automated account is not shown]

**Comment by air2** [comment]
To clarify, if you want the result to be readonly, you could (and IMHO should) do:
```
const source: ReadonlyArray<string> = ['data']
const destination: ReadonlyArray<string> =  cloneDeep(source)
```

**Comment by cdauth** [comment]
Just want to add a reference to the (closed) issue where this was mentioned before: #32066
